src/devices/bus/mononcol/slot.h

```diff
@@ -0,0 +1,194 @@
+
+class device_mononcol_cart_interface : public device_interface
+{
+public:
+
+	// construction/destruction
+	virtual ~device_mononcol_cart_interface();
+
+	// reading and writing
+	virtual void rom_alloc(u32 size, int width, endianness_t end, char const *tag);
+
+	u8 *get_rom_base()  { return m_rom; }
+	u32 get_rom_size() { return m_rom_size; }
+
+	u8 *get_region_base()  { return m_region.found() ? m_region->base() : nullptr; }
+	u32 get_region_size() { return m_region.found() ? m_region->bytes() : 0U; }
+
+	virtual uint8_t read() { return 0x00; }
+	virtual DECLARE_WRITE_LINE_MEMBER(dir_w) { }
+	virtual void write(uint8_t data) { };
+	virtual void set_ready() { };
+
+	virtual void set_spi_region(uint8_t* region) { }
+	virtual void set_spi_size(size_t size) { }
+
+protected:
+	device_mononcol_cart_interface(machine_config const &mconfig, device_t &device);
+
+	// this replaces m_rom for non-user configurable carts!
+	optional_memory_region  m_region;
+
+	// internal state
+	u8 *m_rom;
+	u32 m_rom_size;
+};
+
+
+enum
+{
+	MONONCOL_ROM8_WIDTH = 1,
+};
+
+#define MONONCOL_ROM_REGION_TAG ":cart:rom"
+
+class mononcol_slot_device : public device_t,
+								public device_rom_image_interface,
+								public device_single_card_slot_interface<device_mononcol_cart_interface>
+{
+public:
+	virtual ~mononcol_slot_device();
+
+	template <typename... T> void set_device_load(T &&... args) { m_device_image_load.set(std::forward<T>(args)...); }
+	template <typename... T> void set_device_unload(T &&... args) { m_device_image_unload.set(std::forward<T>(args)...); }
+
+	void set_interface(char const *interface) { m_interface = interface; }
+	void set_default_card(char const *def) { m_default_card = def; }
+	void set_extensions(char const *exts) { m_extensions = exts; }
+	void set_width(int width) { m_width = width; }
+	void set_endian(endianness_t end) { m_endianness = end; }
+
+	// device_image_interface implementation
+	virtual image_init_result call_load() override;
+	virtual void call_unload() override;
+	virtual bool is_reset_on_load() const noexcept override { return true; }
+	virtual char const *image_interface() const noexcept override { return m_interface; }
+	virtual char const *file_extensions() const noexcept override { return m_extensions; }
+
+	// slot interface overrides
+	virtual std::string get_default_card_software(get_default_card_software_hook &hook) const override;
+
+	u32 common_get_size(char const *region);
+	void common_load_rom(u8 *ROM, u32 len, char const *region);
+
+	virtual void rom_alloc(u32 size, int width, endianness_t end) { if (m_cart) m_cart->rom_alloc(size, width, end, tag()); }
+
+	u8* get_rom_base()
+	{
+		if (!m_cart)
+			return nullptr;
+		else if (!user_loadable())
+			return m_cart->get_region_base();
+		else
+			return m_cart->get_rom_base();
+	}
+	u32 get_rom_size()
+	{
+		if (!m_cart)
+			return 0U;
+		else if (!user_loadable())
+			return m_cart->get_region_size();
+		else
+			return m_cart->get_rom_size();
+	}
```

Comment:
You don’t need the `user_loadable` check given this is always for a physical cartridge slot.